Automattic/wp-calypso
# New Checkout: split purchase details into second column

This does a few things with the ultimate goal of splitting out a summary into a separate column on desktop screens.

- adds the floated `CheckoutSummary` component to composite-checkout (package, demo, documentation)
- moves `CheckoutStepBody` component inside `CheckoutSteps`
- uses `CheckoutSummary` in WPcom's implementation of composite-checkout
- moves the current site's address to the review step
- adds a default set of purchase features to the summary (per-product features coming in a future PR)
- hides `CheckoutSummary` on mobile screens (collapse mobile toggle coming in future PR)

Before | After
------------ | -------------
<img width="1164" alt="before-desktop" src="https://user-images.githubusercontent.com/942359/79905306-05b8ca00-83e4-11ea-9c19-7f66901e6e2e.png"> | <img width="1164" alt="after-desktop" src="https://user-images.githubusercontent.com/942359/79905305-05203380-83e4-11ea-8692-4fd43aee5769.png">
<img width="374" alt="before-mobile" src="https://user-images.githubusercontent.com/942359/79905302-05203380-83e4-11ea-9225-1bca7db43ff0.png"> | <img width="372" alt="after-mobile" src="https://user-images.githubusercontent.com/942359/79905300-04879d00-83e4-11ea-936b-54c262a6ae4e.png">

Fixes: #41088

**To test:**
- it might help to step through the PR, commit by commit, but no promises that I didn't change my mind later on in the PR
- run the composite-checkout demo with `yarn run composite-checkout-demo`
- verify that minimal CheckoutSummary is displayed
- verify that the demo otherwise works as expected
- from dev or calypso.live, and a plan and/or domain to your cart and visit Checkout with `?flags=composite-checkout-testing`
- verify that on desktop (`>960px`), the checkout summary is displayed, floated to the right, with a list of default features, and the tax (once calculated) and total
- verify that on screens smaller than desktop, the checkout summary is hidden
- verify that Checkout otherwise works as expected
